src/test/java/hudson/plugins/jira/JiraBuildActionTest.java

```diff
@@ -1,38 +1,47 @@
 package hudson.plugins.jira;
 
-import static org.mockito.Mockito.mock;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertEquals;
 
-import java.util.ArrayList;
-
-import hudson.plugins.jira.model.JiraIssue;
+import hudson.model.Job;
+import hudson.model.Run;
+import hudson.util.XStream2;
+import io.jenkins.plugins.casc.misc.JenkinsConfiguredWithCodeRule;
+import org.junit.Rule;
 import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.recipes.LocalData;
 
-import hudson.model.FreeStyleBuild;
-import hudson.plugins.jira.deprecated.DeprecatedJiraBuildAction;
-import hudson.util.XStream2;
 
 /**
- * Test if existing serialized JiraBuildAction information
- * will be loaded after upgrading jira plugin version to with
- * new JiraBuildAction class version introduced in PR-72.
- *
+ * Place needed resources in src/test/resources/compatibility
  */
 public class JiraBuildActionTest {
 
-    @Test
-    public void binaryCompatibility() {
-        XStream2 xStream2 = new XStream2();
-
-        FreeStyleBuild b = mock(FreeStyleBuild.class);
-        DeprecatedJiraBuildAction deprecatedJiraBuildAction = new DeprecatedJiraBuildAction(b, new ArrayList<JiraIssue>());
-
-        String xml = xStream2.toXML(deprecatedJiraBuildAction);
-        System.out.println(xml);
-        xml = xml.replaceAll("hudson.plugins.jira.deprecated.DeprecatedJiraBuildAction", "hudson.plugins.jira.JiraBuildAction");
-        Object fromXML = xStream2.fromXML(xml);
-        System.out.println(fromXML.getClass().getName());
-        JiraBuildAction jiraBuildAction = (JiraBuildAction) fromXML;
-        System.out.println(jiraBuildAction.getDisplayName());
-    }
+  @Rule
+  public JenkinsRule r = new JenkinsConfiguredWithCodeRule();
+
+
+  /**
+   * Test if existing serialized JiraBuildAction information will be loaded after upgrading jira
+   * plugin version to with new JiraBuildAction class version introduced in PR-72.
+   */
+  @Test
+  @LocalData
+  public void binaryCompatibility() {
+    assertEquals("Jenkins JiraBuildActionTest config", r.jenkins.getSystemMessage());
+
+    Job job = r.getInstance().getItemByFullName("/project", Job.class);
+    Run run = job.getBuildByNumber(2);
+    assertEquals("job/project/2/", run.getUrl());
+
+    XStream2 xStream2 = new XStream2();
+    String xml = xStream2.toXML(run);
+    System.out.println(xml);
+
+    JiraBuildAction jba = run.getAction(JiraBuildAction.class);
```

Comment:
this throws NPE, that's why I was looking at xml of `run`

Reply:
Throws NPE from what?

Reply:
Just from getAction() method:
```
java.lang.NullPointerException
	at hudson.plugins.jira.JiraBuildActionTest.binaryCompatibility(JiraBuildActionTest.java:55)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.jvnet.hudson.test.JenkinsRule$1.evaluate(JenkinsRule.java:599)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:288)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:282)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

Reply:
Line number 55 is checking `jba.owner`, are you sure?

Reply:
Exactly, jba is null in my debug session..